diff -Naur lpeg-0.12.orig/lptree.c lpeg-0_12/lptree.c
--- lpeg-0.12.orig/lptree.c	Fri Apr 12 09:31:19 2013
+++ lpeg-0_12/lptree.c	Fri Aug  2 10:15:43 2013
@@ -1217,8 +1217,7 @@
 };
 
 
-int luaopen_lpeg (lua_State *L);
-int luaopen_lpeg (lua_State *L) {
+LUALIB_API int luaopen_lpeg (lua_State *L) {
   luaL_newmetatable(L, PATTERN_T);
   lua_pushnumber(L, MAXBACK);  /* initialize maximum backtracking */
   lua_setfield(L, LUA_REGISTRYINDEX, MAXSTACKIDX);
